Be more explicit in the @odata.Type warning (#558)
cap-js/docs: advanced/odata.md

```

::: warning

> This annotation affects the client side facing API only, there is no automatic data modification of any kind (like rounding, truncation, conversion, etc.) 
> behind the scenes. It is in your responsibility to perform all required modifications on the data stream such that the values match their type in the API. 
> If you are not doing the required conversions, it is possible to "cast" any scalar CDS type into any (in-)compatible EDM type:

Reviewer: ```suggestion
This annotation affects the client side facing API only. There's no automatic data modification of any kind behind the scenes, like rounding, truncation, conversion, and so on. It's your responsibility to perform all required modifications on the data stream such that the values match their type in the API. 
If you are not doing the required conversions, you can "cast" any scalar CDS type into any incompatible EDM type:
```

Reviewer: The last sentence, as the introduction to the following snippets, implies that the following example is a **Don't** example. Is that correct?

Author: yes it is
